reactor-core/src/main/java/reactor/util/context/ContextN.java

```diff
@@ -29,42 +30,45 @@
 
 @SuppressWarnings("unchecked")
 final class ContextN extends HashMap<Object, Object>
-		implements Context, Function<Entry<Object, Object>, Entry<Object, Object>> {
+		implements Context, Function<Entry<Object, Object>, Entry<Object, Object>>,
+		           BiConsumer<Object, Object> {
 
 	ContextN(Object key1, Object value1, Object key2, Object value2,
 			Object key3, Object value3, Object key4, Object value4,
 			Object key5, Object value5, Object key6, Object value6) {
 		super(6, 1f);
-		innerPut(key1, value1);
-		innerPut(key2, value2);
-		innerPut(key3, value3);
-		innerPut(key4, value4);
-		innerPut(key5, value5);
-		innerPut(key6, value6);
+		//accept below stands in for "inner put"
+		accept(key1, value1);
+		accept(key2, value2);
+		accept(key3, value3);
+		accept(key4, value4);
+		accept(key5, value5);
+		accept(key6, value6);
 	}
 
 	ContextN(Map<Object, Object> map, Object key, Object value) {
 		super(map.size() + 1, 1f);
-		innerPutAll(map);
-		innerPut(key, value);
+		Objects.requireNonNull(map, "map")
+		       .forEach(this);
+		accept(key, value); //innerPut
 	}
 
 	ContextN(Map<Object, Object> sourceMap, Map<?, ?> other) {
 		super(sourceMap.size() + other.size(), 1f);
```

Comment:
access to probably null value 
both bellow validations are useless - they will never work but just allocate message Strings

Reply:
FYI constant strings are stored in the constant pool and are not allocated every time

Reply:
I'll follow up with a polish moving this part one step above